src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -1749,7 +1750,30 @@ public Double execute(Jedis connection) {
       }
     }.runBinary(key);
   }
+  
+  @Override
+  public ScanResult<byte[]> scan(final byte[] cursor, final ScanParams params) {
+
+    String matchPattern = null;
+
+    if (params == null || (matchPattern = params.match()) == null || matchPattern.isEmpty()) {
+      throw new IllegalArgumentException(JedisCluster.class.getSimpleName() + " only supports scan requests with non-empty match patterns");
+    }
 
+    if (JedisClusterHashTagUtil.isClusterCompliantMatchPattern(matchPattern)) {
+
+      return new JedisClusterCommand< ScanResult<byte[]>>(connectionHandler,
+              maxRedirections) {
+        @Override
+        public ScanResult<byte[]> execute(Jedis connection) {
+          return connection.scan(cursor, params);
+        }
+      }.runBinary(SafeEncoder.encode(matchPattern));
+    } else {
+      throw new IllegalArgumentException(JedisCluster.class.getSimpleName() + " only supports scan requests with match patterns of the following format : '{<HASH_TAG>}*' where <HASH_TAG> is a tag string of your choice");
```

Comment:
@HeartSaVioR  done , I actually did forget to fix those after disposing of the regex pattern matching